mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulCreateTournamentScreen.kt

```diff
@@ -0,0 +1,134 @@
+package ch.epfl.sdp.mobile.state
+
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.core.text.isDigitsOnly
+import ch.epfl.sdp.mobile.application.authentication.AuthenticatedUser
+import ch.epfl.sdp.mobile.application.tournaments.TournamentFacade
+import ch.epfl.sdp.mobile.application.tournaments.TournamentReference
+import ch.epfl.sdp.mobile.ui.i18n.LocalizedStrings
+import ch.epfl.sdp.mobile.ui.tournaments.CreateDialog
+import ch.epfl.sdp.mobile.ui.tournaments.CreateDialogState
+import ch.epfl.sdp.mobile.ui.tournaments.CreateDialogState.*
+import kotlin.math.log2
+import kotlin.math.pow
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+@Composable
+fun StatefulCreateTournamentScreen(
+    user: AuthenticatedUser,
+    navigateToTournament: (reference: TournamentReference) -> Unit,
+    cancelClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+  val scope = rememberCoroutineScope()
+
+  val tournamentFacade = LocalTournamentFacade.current
+  val strings = LocalLocalizedStrings.current
+
+  val state =
+      remember(user, navigateToTournament, cancelClick, scope, tournamentFacade) {
+        ActualCreateTournamentScreenState(
+            user = user,
+            navigateToTournament = navigateToTournament,
+            cancelClick = cancelClick,
+            tournamentFacade = tournamentFacade,
+            strings = strings,
+            scope = scope,
+        )
+      }
+  CreateDialog(
+      state = state,
+      modifier = modifier,
+      poolSizeKey = { it.name },
+      eliminationRoundKey = { it.name },
```

Comment:
Actually, I'd use `it.value` which is the real "key" of the items.

Reply:
Fixed in https://github.com/epfl-SDP/android/pull/336/commits/ce6a4153003b08efbcc80fb1c684b948b8dfb481